config/ompi_setup_cxx.m4

```diff
@@ -29,11 +29,33 @@ AC_DEFUN([OMPI_SETUP_CXX_BANNER],[
 ])
 
 AC_DEFUN([OMPI_PROG_CXX],[
-    OPAL_VAR_SCOPE_PUSH([ompi_cxxflags_save])
+    OPAL_VAR_SCOPE_PUSH([ompi_cxxflags_save ompi_cxx_argv0])
+
     ompi_cxxflags_save="$CXXFLAGS"
     AC_PROG_CXX
     AC_PROG_CXXCPP
     CXXFLAGS="$ompi_cxxflags_save"
+
+    # Note: according to the Autoconf docs, if no C++ compiler is
+    # found, $CXX is still set to "g++" (!!).  So make sure that we
+    # actually found a C++ compiler; if not, set CXX to "no", per
+    # thread at
+    # https://www.open-mpi.org/community/lists/users/2013/02/21356.php,
+    # which advises us to set Libtool precious variables to "no" if we
+    # don't want Libtool to setup that language at all.
+    AS_IF([test "x$CXX" = "x"], [CXX=no])
+    set dummy $CXX
+    ompi_cxx_argv0=[$]2
+    OPAL_WHICH([$ompi_cxx_argv0], [OMPI_CXX_ABSOLUTE])
+    AS_IF([test "x$OMPI_CXX_ABSOLUTE" = "x"],
```

Comment:
Do we really need OMPI_CXX_ABSOLUTE?  It looks like the only place it is used is in ompi_info and oshmem_info, which seems like a really bad reason to keep carrying this code around.

Reply:
Fair enough (I didn't track down where it was used).  I can ditch it.

Reply:
Actually, I now remember why this was added: we had user complaints about `ompi_info` showing just `g++` or `icpc` or whatever the basename of the C++ compiler was.  But that didn't help users track which *which* C++ compiler was used (if they have several different versions of `g++` or `icpc` [or whatever] available, e.g., depending on which environment module was loaded, etc.).

Having `ompi_info` emit the absolute path name to the compiler used was the way to disambiguate that.

Do you feel strongly about removing it?  I think it's kinda useful.

Reply:
I think it has no use today.  mpic++ won't call /usr/bin/g++, it will call g++.  And since that's the only place that will still invoke the c++ compiler, it's really giving customers a false sense of security.

Reply:
@bwbarrett and I discussed in Slack.  We decided that this is "good enough" for this PR.  He'll make a new PR that does a Better Thing: it'll set the ABSOLUTE values (probably in `ompi/tools/wrappers/Makefile.am`?) at build time, so that we actually know for sure which compilers were used to build Open MPI.